### PR TITLE
feat: Improved table loading

### PR DESCRIPTION
**Pull request by kasperkristensen**
**What**
- Adds a new `TableContainer` component responsible for handling the loading state of tables and table pagination, as these two concepts are tightly related.
- Adds new `Skeleton` and `SkeletonProvider` that is used to display loading skeletons and control their state.
- Fixes broke search on all main tables, the ones seen when navigating to /products, /orders, etc., and ´SelectableTables´
- Fixes bug that would occur when (working) table search occurs inside a LayeredModal, which would result in a max depth error. The issue was solved by memorizing the data source, and I haven't been able to reproduce the issue after the change. **But if you see otherwise during testing please shout!**

**How**
- The new `TableContainer` solves three issues:
  1. It reduces layout shifts by adding a calculated min-height to tables, defaults to 12 but can be manually set using `numberOfRows`.
  2. Fixes invalid use of nesting a spinner inside a table while it is loaded. Instead, the TableContainer accepts an `isLoading` param, and while this is true it renders a spinner on top of the table.
  3. Hides page number and count flickering in table pagination while `isLoading` is true.
- The table search has been fixed in several ways:
  1. Reset `offset` when `q` is updated. To test paginate a couple of times, enter a search query, and see that the pagination/offset is reset. **If I've missed somewhere please point it out**.
  2. Make search work in `SelectableTable`, and update typing.

Resolves CORE-777, CORE-765, CORE-387

### CODE REVIEW

File: src/components/organisms/table-container/index.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+const TableContainer = <T extends boolean>({
+  children,
+  hasPagination,
+  pagingState,
+  isLoading,
+  numberOfRows = 12,
+}: Props<T>) => {
+  // We use the number of rows (query limit) plus the header row to calculate the minimum height of the table, to avoid the table jumping around while loading.
+  const minHeight = (numberOfRows + 1) * ROW_HEIGHT
+
```
Comment (srindom) [inline src/components/organisms/table-container/index.tsx:24]: praise: 😍

**Review by srindom — APPROVED** [review]
LGTM!

**Review by olivermrbl — APPROVED** [review]
This is epic - great work @kasperkristensen !

**Review by adrien2p — COMMENTED** [review]
I know I am a bit late 😂 LGTM but one comment though, I would like your opinion

File: src/components/templates/collection-product-table/add-product-table.tsx [hunk 2]
```diff
@@ -160,51 +160,48 @@ const AddProductsTable: React.FC<AddProductsTableProps> = ({
           <h3 className="inter-xlarge-semibold">Add Products</h3>
         </Modal.Header>
         <Modal.Content>
-          <div className="w-full flex flex-col justify-between h-[650px]">
+          <TableContainer
+            hasPagination
+            numberOfRows={PAGE_SIZE}
+            isLoading={isLoading}
+            pagingState={{
+              count: count!,
```
[the remark this answers is not in the document]
Comment (kasperkristensen) [reply src/components/templates/collection-product-table/add-product-table.tsx:168]: It does not really matter, as the count is hidden behind a skeleton while `isLoading` is true, so you won't be able to see whether it says 0 or undefined. The same goes for the number of pages. Also, I've added `keepPreviousData` to all of the queries so the count and page count won't flash in between page changes, as that data is mostly static (unless someone for example added a new product in between page changes).

[the remark this answers is not in the document]
Comment (adrien2p) [reply src/components/templates/collection-product-table/add-product-table.tsx:168]: thanks for the explanation :) make a lot of sense

**Pull request merged**